From 048a223f875f8123ab838ec526f3fc6e91bfe9aa Mon Sep 17 00:00:00 2001
From: AnNDP <AnNDP@fsoft.com.vn>
Date: Wed, 6 Jun 2018 03:51:57 -0400
Subject: [PATCH 8/9] waylandsink: Change the color format mapping for XBGR8888
 in wl_kms

This change fixes the format interpretation difference between
GStreamer and the wayland server side in wl_kms.
The mapping for BGRX8888 is also changed to keep symmetry of
the interpretation.

Signed-off-by: AnNDP <AnNDP@fsoft.com.vn>
---
 ext/wayland/wlvideoformat.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/ext/wayland/wlvideoformat.c b/ext/wayland/wlvideoformat.c
index 77cfac8..6ee1578 100644
--- a/ext/wayland/wlvideoformat.c
+++ b/ext/wayland/wlvideoformat.c
@@ -46,18 +46,18 @@ static const wl_ShmVideoFormat shm_formats[] = {
 #if G_BYTE_ORDER == G_BIG_ENDIAN
   {WL_SHM_FORMAT_XRGB8888, GST_VIDEO_FORMAT_xRGB},
   {WL_SHM_FORMAT_ARGB8888, GST_VIDEO_FORMAT_ARGB},
-  {WL_SHM_FORMAT_XBGR8888, GST_VIDEO_FORMAT_xBGR},
+  {WL_SHM_FORMAT_XBGR8888, GST_VIDEO_FORMAT_BGRx},
   {WL_SHM_FORMAT_RGBX8888, GST_VIDEO_FORMAT_RGBx},
-  {WL_SHM_FORMAT_BGRX8888, GST_VIDEO_FORMAT_BGRx},
+  {WL_SHM_FORMAT_BGRX8888, GST_VIDEO_FORMAT_xBGR},
   {WL_SHM_FORMAT_ABGR8888, GST_VIDEO_FORMAT_ABGR},
   {WL_SHM_FORMAT_RGBA8888, GST_VIDEO_FORMAT_RGBA},
   {WL_SHM_FORMAT_BGRA8888, GST_VIDEO_FORMAT_BGRA},
 #else
   {WL_SHM_FORMAT_XRGB8888, GST_VIDEO_FORMAT_BGRx},
   {WL_SHM_FORMAT_ARGB8888, GST_VIDEO_FORMAT_BGRA},
-  {WL_SHM_FORMAT_XBGR8888, GST_VIDEO_FORMAT_RGBx},
+  {WL_SHM_FORMAT_XBGR8888, GST_VIDEO_FORMAT_xRGB},
   {WL_SHM_FORMAT_RGBX8888, GST_VIDEO_FORMAT_xBGR},
-  {WL_SHM_FORMAT_BGRX8888, GST_VIDEO_FORMAT_xRGB},
+  {WL_SHM_FORMAT_BGRX8888, GST_VIDEO_FORMAT_RGBx},
   {WL_SHM_FORMAT_ABGR8888, GST_VIDEO_FORMAT_RGBA},
   {WL_SHM_FORMAT_RGBA8888, GST_VIDEO_FORMAT_ABGR},
   {WL_SHM_FORMAT_BGRA8888, GST_VIDEO_FORMAT_ARGB},
-- 
1.9.1

